datetime.go

```diff
@@ -0,0 +1,85 @@
+// +build darwin ios
```

Comment:
I went a bit overboard with the date conversion code here, but it might come in handy for other things.